src/NCPoly.jl

```diff
@@ -18,6 +18,32 @@ function is_exact_type(a::Type{T}) where {S <: NCRingElem, T <: NCPolyRingElem{S
    return is_exact_type(S)
 end
 
+@doc md"""
+    dense_poly_type(::Type{T}) where T<:NCRingElement
+
+The type of multivariate polynomials with coefficients of type `T`.
+Falls back to `Generic.NCPoly{T}` respectively `Generic.Poly{T}`.
+
+# Examples
+```julia
+dense_poly_type(typeof(ZZ()))
+```
+"""
+dense_poly_type(::Type{T}) where T<:NCRingElement = Generic.NCPoly{T}
+
+@doc md"""
+    poly_ring_type(::Type{T}) where T<:NCRing
+
+The type of polynomial rings with coefficients of type `T`.
+Implemented via [`dense_poly_type`](@ref).
```

Comment:
Wouldn't it then make more sense to call it `dense_poly_ring_type`?

Reply:
Maybe, on the other hand, we settled on not spelling out `dense` during the big renaming discussion.

Reply:
Lets continue in #1289